### PR TITLE
TINY-10330: add check to `BespokeNumberInput` to avoid `updateInputValue` if the component is not connect

### PR DESCRIPTION
**Pull request by lorenzo-pomili**
Related Ticket: TINY-10330

Description of Changes:

- [x] make check in `BespokeNumberInput` to avoid `updateInputValue` if the component is not connected
- [x] remove the double call of `setToolbar` from `iframe.ts`
- [x] add `chai-spies` to test that the warning is not showed
- [x] add `fontsizeinput` to fullDemo page

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] ~Docs ticket created (if applicable)~

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts [hunk 1]
```diff
@@ -418,6 +421,19 @@ describe('browser.tinymce.themes.silver.throbber.NumberInputTest', () => {
     TinyAssertions.assertContent(editor, `<p style="font-size: ${originalFontSize};">a<span style="font-size: ${originalFontSize};">b</span>c</p>`);
   });
 
+  it('TINY-10330: the presence and updates of fontsizeinput should not trigger warnings', async () => {
+    chai.use(spies);
+    const spy = chai.spy.on(console, 'warn');
```
Comment (spocke) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts:426]: I wonder if we will be using this enough to warrant a library to have to bump etc since this is the first time we are in the need of this. We could just do a spy thing locally in this test just patch and un-patch the console.warn function?

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts [hunk 2]
```diff
@@ -25,8 +25,10 @@ const createBespokeNumberInput = (editor: Editor, backstage: UiFactoryBackstage,
   const onSetup = onSetupEvent(editor, 'NodeChange SwitchMode', (api: BespokeSelectApi) => {
     const comp = api.getComponent();
     currentComp = Optional.some(comp);
-    spec.updateInputValue(comp);
-    Disabling.set(comp, !editor.selection.isEditable());
+    if (comp.getSystem().isConnected()) {
```
Comment (spocke) [inline modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts:28]: Why was this needed is this being executed to early?

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts (hunk 1 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts:426]: I tried to refactor in this way:
```
// eslint-disable-next-line no-console
    const oldConsoleWarn = console.warn;
    let isWarnCalled = false;
    // eslint-disable-next-line no-console
    console.warn = (a) => {
      isWarnCalled = true;
      oldConsoleWarn(a);
    };

    const editor = hook.editor();
    editor.setContent('<p style="font-size: 20px;">abc</p>');
    TinySelections.setSelection(editor, [ 0, 0 ], 1, [ 0, 0 ], 2);
    const input = TinyUiActions.clickOnToolbar<HTMLInputElement>(editor, '.tox-number-input input');
    await Waiter.pTryUntilPredicate(`Wait for the new input value is setted`, () => input.dom.value === '20px');

    expect(isWarnCalled).to.eq(false);
    // eslint-disable-next-line no-console
    console.warn = oldConsoleWarn;
```

and it seems to work, we could change in this way or add the library so we can se it also for the future tests

I this the lib could be useful but I don't have strong opinion on it

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts (hunk 2 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts:28]: actually the problem seems to be that the component is composed by 3 components (minus, input and plus) but the "un-mounting" works only for the last one :( , so in our case the `plus`, I'm trying to figure out why is that :/

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts [hunk 3]
```diff
@@ -418,6 +420,31 @@ describe('browser.tinymce.themes.silver.throbber.NumberInputTest', () => {
...
 
+  it('TINY-10330: the presence and updates of fontsizeinput should not trigger warnings', async () => {
+    // eslint-disable-next-line no-console
+    const storedConsoleWarn = console.warn;
+    const warnings: string[] = [];
+
+    // eslint-disable-next-line no-console
+    console.warn = (a) => {
+      // eslint-disable-next-line no-console
+      console.log('a: ', a);
```
Comment (spocke) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts:431]: Maybe call `storedConsoleWarn` here if that is possible?

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts (hunk 2 above)
Comment (spocke) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts:28]: That would explain why things are not working as expected when the setToolbar is executed twice. This could produce issues if the fontsizeinput is in the more drawer of the toolbar so that it gets rendered and re-rendered over and over so we need to fix this.

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts (hunk 3 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/NumberInputTest.ts]: thanks, I forget that `console.log` :sweat_smile:

File: modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts (hunk 2 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts]: ok, I implemented the fix that you suggested thanks for the help :)

File: modules/tinymce/CHANGELOG.md [hunk 4]
```diff
@@ -36,6 +36,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Removed use of `async` for editor rendering which caused visual blinking when reloading the editor in-place. #TINY-10249
 - Hex colors are no longer always converted to RGB. #TINY-9819
 - Merging an external `p` inside a `list` via delete or backspace would incorrectly try to move a parent element inside a child element. #TINY-10289
+- The `fontsizeinput` triggered a console warning from `NoContextApi`. #TINY-10330
```
Comment (spocke) [inline modules/tinymce/CHANGELOG.md:39]:
```suggestion
- The `fontsizeinput` toolbar item was causing console warnings when toolbar items were clicked. #TINY-10330
```

**Review by spocke — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 4 above)
Comment (lorenzo-pomili) [reply modules/tinymce/CHANGELOG.md]: changed

**Review by hamza0867 — APPROVED** [review]
